src/main/config.cpp

```diff
@@ -153,6 +154,10 @@ void DBConfig::SetOption(DatabaseInstance *db, const ConfigurationOption &option
 	option.set_global(db, *this, input);
 }
 
+void DBConfig::SetCustomBufferManager(unique_ptr<BufferManager> buffer_manager) {
```

Comment:
Maybe we need to protect this from being able to be run after the buffer manager has already been used? database is already created etc..